core/processor/ProcessorParseJsonNative.cpp

```diff
@@ -60,19 +90,19 @@ bool ProcessorParseJsonNative::ProcessEvent(const StringView& logPath, PipelineE
 
     auto rawContent = sourceEvent.GetContent(mSourceKey);
 
-    bool res = true;
-    res = JsonLogLineParser(sourceEvent, logPath, e);
+    bool parseSuccess = true;
+    parseSuccess = JsonLogLineParser(sourceEvent, logPath, e);
 
-    if (!res && !mDiscardUnmatch) {
-        AddLog(LogParser::UNMATCH_LOG_KEY, // __raw_log__
+    if (!parseSuccess && (mKeepingSourceWhenParseFail || mCopingRawLog)) {
+        AddLog(UNMATCH_LOG_KEY, // __raw_log__
                rawContent,
                sourceEvent); // legacy behavior, should use sourceKey
     }
-    if (res || !mDiscardUnmatch) {
-        if (mUploadRawLog && (!res || !mRawLogTagOverwritten)) {
-            AddLog(mRawLogTag, rawContent, sourceEvent); // __raw__
+    if (parseSuccess || mKeepingSourceWhenParseFail) {
+        if (mKeepingSourceWhenParseSucceed && (!parseSuccess || !mRawLogTagOverwritten)) {
+            AddLog(mRenamedSourceKey, rawContent, sourceEvent); // __raw__
         }
-        if (res && !mSourceKeyOverwritten) {
+        if (parseSuccess && !mSourceKeyOverwritten) {
             sourceEvent.DelContent(mSourceKey);
         }
         return true;
```

Comment:
太绕了，简单点
```suggestion
    if ((parseSuccess && mKeepingSourceWhenParseSucceed || !parseSuccess && mKeepingSourceWhenParseFail) && mSourceKey != mRenamedSourceKey) {
        if (!mSourceKeyOverwritten) {
            sourceEvent.DelContent(mSourceKey);
        }
        AddLog(mRenamedSourceKey, rawContent, sourceEvent, false);
    }
    // legacy behavior
    if (!parseSuccess && mKeepingSourceWhenParseFail && mCopingRawLog) {
        AddLog(UNMATCH_LOG_KEY, rawContent, sourceEvent, false);
        return true;
```
另外，这段逻辑也应该整体打包函数放到CommonParserOptions里，各个地方复用